--- app-vulnerable.js	Mon Sep 23 06:11:58 2013
+++ app.js	Mon Sep 23 06:14:55 2013
@@ -39,7 +39,7 @@
     } catch (e) {}
   },
   serialize: function(session) {
-    return 'session=' + Buffer(JSON.stringify(session)).toString('base64');
+    return 'session=' + Buffer(JSON.stringify(session)).toString('base64') + '; HttpOnly';
   },
   clear: 'session=; Expires=Thu, 01 Jan 1970 00:00:00 GMT'
 };
