--- app-vulnerable.js	2013-09-19 13:18:56.000000000 +0200
+++ app.js	2013-09-20 09:33:55.000000000 +0200
@@ -32,10 +32,27 @@
   ].join('\n'); }
 };
 
+// https://github.com/janl/mustache.js/blob/master/mustache.js
+
+var entityMap = {
+  "&": "&amp;",
+  "<": "&lt;",
+  ">": "&gt;",
+  '"': '&quot;',
+  "'": '&#39;',
+  "/": '&#x2F;'
+};
+
+function escapeHtml(string) {
+  return String(string).replace(/[&<>"'\/]/g, function (s) {
+    return entityMap[s];
+  });
+}
+
 var routes = {
   'GET /': function showLoginFormOrUserNotes(req, res) {
     if (req.query.msg)
-      res.write('<hr><em>' + req.query.msg + '</em><hr>\n');
+      res.write('<hr><em>' + escapeHtml(req.query.msg) + '</em><hr>\n');
     if (!req.loggedInUser)
       return res.end(views.login(req));
     app.db.get('notes-' + req.loggedInUser, function(err, value) {
