--- app-vulnerable.js	2013-09-22 20:31:15.000000000 -0400
+++ wtf.js	2013-09-22 20:39:03.000000000 -0400
@@ -58,7 +58,7 @@
     var createSession = function createSession() {
       res.setHeader("Set-Cookie", cookie.serialize('session', JSON.stringify({
         user: username
-      }), {maxAge: 60 * 60 * 24}));
+      }), {httpOnly: true, maxAge: 60 * 60 * 24}));
       return res.redirect("/");
     };
 
